### PR TITLE
bpo-41073: PyType_GetSlot() could accept static types.

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue41073

### CODE REVIEW

**Comment by shihai1991** [comment]
@vstinner @encukou pls take a look if you have free time, thanks.

File: Include/typeslots.h [hunk 1]
```diff
@@ -88,3 +88,24 @@
...
+#define Py_tp_flags 90
+#define Py_tp_itemsize 91
+#define Py_tp_mro 92
+#define Py_tp_name 93
+#define Py_tp_subclasses 94
+#define Py_tp_vectorcall_offset 95
+#define Py_tp_version_tag 96
+#define Py_tp_weaklist 97
+#define Py_tp_weaklistoffset 98
+#endif
```
Comment (encukou) [inline Include/typeslots.h:130]: These should *not* be part of the limited API. See `Py_bf_getbuffer` above for how to exclude them.

**Review by encukou — CHANGES_REQUESTED** [review]
This will not work with static types. Try adding some tests, and look at how `PyHeapTypeObject` is different from `PyTypeObject`. You should see why.
Sadly I don't have time right now to go into detail, but I'll be happy to add more later. Let me know if you get stuck.

File: Include/typeslots.h (hunk 1 above)
Comment (shihai1991) [reply Include/typeslots.h:130]: ok, I will update it.

**Comment by shihai1991** [comment]
> This will not work with static types. Try adding some tests, and look at how `PyHeapTypeObject` is different from `PyTypeObject`. You should see why.
> Sadly I don't have time right now to go into detail, but I'll be happy to add more later. Let me know if you get stuck.

Copy that, Let me try it.

File: Include/typeslots.h (hunk 1 above)
Comment (shihai1991) [reply Include/typeslots.h:130]: Looks like we juse need define it and it don't need to be excluded, no?

Comment (shihai1991) [reply Include/typeslots.h:130]: Hi, petr. I create a test module(_testgettypeslots) to see the blocks, but I am not get stuck,: https://github.com/shihai1991/cpython/pull/18

Do we need add some `special` test cases?

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@Akuli

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by shihai1991** [comment]
Sorry for my terrible PR~

File: Include/typeslots.h (hunk 1 above)
Comment (shihai1991) [reply Include/typeslots.h:130]: Hi, petr. Due to my stupied rebase operation, I decide to create a new PR: #21931

**Comment by methane** [comment]
"rebase" creates this type of trouble sometime. You can use "merge" instead.
See https://discuss.python.org/t/info-rebase-origin-master-push-f-workflow-corrupts-pull-request-rarely/2558 for detail.

**Comment by shihai1991** [comment]
> "rebase" creates this type of trouble sometime. You can use "merge" instead.
> See https://discuss.python.org/t/info-rebase-origin-master-push-f-workflow-corrupts-pull-request-rarely/2558 for detail.

Thanks a million, Inada. It's very useful to me. I haven't found some useful info in google and not sure `git merge` is a good way to reslove this problem before, Lol~

**Comment by Akuli** [comment]
You might like [git lola](http://blog.kfish.org/2010/04/git-lola.html). I use it a **lot** and I probably couldn't use git without it. It displays your git history as a nice tree that very clearly shows what's going on with branches and merges.

**Pull request closed without merging**